pages/api/integrations/thetis/update-user.ts

```diff
@@ -14,48 +14,173 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
+      return res
+        .status(200)
+        .json({ message: `Could not update event type. Thetis user id: ${id} does not exist in Vulcan.` });
+    }
+
+    const eventTypes = await prisma.eventType.findMany({
       where: {
-        hidden: false,
         users: {
           every: {
             thetisId: id,
           },
         },
       },
+      select: {
+        id: true,
+        slug: true,
+      },
+    });
+
+    const DEFAULT_EVENT_TYPE_SLUG = "30min";
+    const DEFAULT_INTRO_EVENT_TYPE_SLUG = `intro-${currentUser.username}`;
+    const INACTIVE_MODIFIER = "inactive";
+
+    const defaultEventTypeActive = eventTypes.find((et) => et.slug === DEFAULT_EVENT_TYPE_SLUG);
+    const defaultEventTypeInactive = eventTypes.find(
+      (et) => et.slug === `${DEFAULT_EVENT_TYPE_SLUG}-${INACTIVE_MODIFIER}`
+    );
+
+    const defaultEventType = defaultEventTypeActive || defaultEventTypeInactive;
+
+    if (!defaultEventType) {
+      logger.error(
+        `Default EventType with slug: ${DEFAULT_EVENT_TYPE_SLUG} does not exist for Vulcan user with id: ${currentUser.id}.\n
+        If Vulcan user has already been onboarded OR Vulcan user hasn't been onboarded but has other event types,
+        manual action is required to create the default event type with slug: ${DEFAULT_EVENT_TYPE_SLUG}.`
+      );
+      return res.status(200).json({
+        message: `A default EventType with slug: ${DEFAULT_EVENT_TYPE_SLUG} does not exist for Vulcan user with id: ${currentUser.id}. See application logs or more information on how to resolve this.`,
+      });
+    }
+
+    const updatedDefaultEventType = await prisma.eventType.update({
+      where: {
+        id: defaultEventType.id,
+      },
       data: {
         description: `1on1 Meeting with ${instructorPublicName}`,
-        price,
+        price: price,
+        slug: isActive ? DEFAULT_EVENT_TYPE_SLUG : `${DEFAULT_EVENT_TYPE_SLUG}-${INACTIVE_MODIFIER}`,
       },
     });
 
-    logger.info(instructorPublicName, instructorHandle);
+    logger.info(
+      `Default EventType updated with id: ${updatedDefaultEventType.id} (slug: ${updatedDefaultEventType.slug}) with price: $${price}.`
+    );
+
+    const defaultIntroEventTypeActive = eventTypes.find((et) => et.slug === DEFAULT_INTRO_EVENT_TYPE_SLUG);
+    const defaultIntroEventTypeInactive = eventTypes.find(
+      (et) => et.slug === `${DEFAULT_INTRO_EVENT_TYPE_SLUG}-${INACTIVE_MODIFIER}`
+    );
+
+    const defaultIntroEventType = defaultIntroEventTypeActive || defaultIntroEventTypeInactive;
+
+    logger.info("defaultIntroEventType", defaultIntroEventType);
+
+    const introEventTypeSlug =
+      isActive && introductoryPrice && introductoryPrice > 0
+        ? DEFAULT_INTRO_EVENT_TYPE_SLUG
+        : `${DEFAULT_INTRO_EVENT_TYPE_SLUG}-${INACTIVE_MODIFIER}`;
+
+    // eslint-disable-next-line
+    // @ts-ignore
+    delete updatedDefaultEventType?.id;
```

Comment:
[Generally preferable to avoid mutating the object](https://blog.sapegin.me/all/avoid-mutation/). Could do

`const updatedDefaultEventTypeWithoutId = { ...updatedDefaultEventType, id: null };`

and then reference the new object going forward. Helps with readability as the code is very explicit by the naming.